presto-main/src/main/java/com/facebook/presto/event/QueryMonitor.java

```diff
@@ -169,6 +171,34 @@ public void queryUpdatedEvent(QueryInfo queryInfo)
         eventListenerManager.queryUpdated(new QueryUpdatedEvent(createQueryMetadata(queryInfo)));
     }
 
+    public void publishQueryProgressEvent(BasicQueryInfo queryInfo)
+    {
+        try {
+            eventListenerManager.publishQueryProgress(new QueryProgressEvent(
+                    new QueryMetadata(
+                            queryInfo.getQueryId().toString(),
+                            queryInfo.getSession().getTransactionId().map(TransactionId::toString),
+                            queryInfo.getQuery(),
+                            queryInfo.getQueryHash(),
+                            queryInfo.getPreparedQuery(),
+                            queryInfo.getState().toString(),
+                            queryInfo.getSelf(),
+                            Optional.empty(),
+                            Optional.empty(),
+                            Optional.empty(),
+                            Optional.empty(),
+                            ImmutableList.of(),
+                            queryInfo.getSession().getTraceToken()),
+                    createQueryStatistics(queryInfo),
+                    createQueryContext(queryInfo.getSession(), queryInfo.getResourceGroupId()),
+                    queryInfo.getQueryType(),
+                    ofEpochMilli(queryInfo.getQueryStats().getCreateTime().getMillis())));
+        }
+        catch (Throwable e) {
+            log.error(e, "Error sending Query Progress Event");
```

Comment:
I don't think we need try-catch here, because any issues should be handled by the specific EventListener implementation which does not throw checked exception. The try-catch should happen there.